src/test/java/jsr223/perl/PerlScriptEngineFactoryTest.java

```diff
@@ -26,140 +26,77 @@
-        new PerlScriptEngineFactory(new PerlProcessBuilderUtilities(), null);
-    }
-
-    @Test(expected = NullPointerException.class)
-    public void testThatProcessBuilderCannotBeNull() {
-        new PerlScriptEngineFactory(null, new PerlVersionGetter(new PerlProcessBuilderUtilities()));
-    }
-
-    @Test(expected = NullPointerException.class)
-    public void testThatVersionGetterAndProcessBuilderCannotBeNull() {
-        new PerlScriptEngineFactory(null, null);
-    }
+    PerlScriptEngineFactory perlScriptEngineFactory = new PerlScriptEngineFactory();
 
     @Test
     public void testThatPerlVersionGetterIsUsed() {
-        PerlVersionGetter perlVersionGetterMock = mock(PerlVersionGetter.class);
-        String testVersion = "someVersion 44";
-        when(perlVersionGetterMock.getPerlVersion(any(PerlProcessBuilderFactory.class))).thenReturn(testVersion);
-
-        PerlScriptEngineFactory perlScriptEngineFactory = new PerlScriptEngineFactory(new PerlProcessBuilderUtilities(),
-                                                                                      perlVersionGetterMock);
-
-        assertThat(perlScriptEngineFactory.getLanguageVersion(), is(testVersion));
-    }
-
-    @Test
-    public void testThatDefaultEngineVersionIsReturned() {
-        PerlScriptEngineFactory perlScriptEngineFactory = new PerlScriptEngineFactory();
-        assertThat(perlScriptEngineFactory.getEngineVersion(), is("0.1.0"));
+        String version = perlScriptEngineFactory.getLanguageVersion();
+        assertThat(version, notNullValue());
     }
 
     @Test
     public void testThatPerlScriptEngineFactoryReturnsScriptEngine() {
-        PerlScriptEngineFactory perlScriptEngineFactory = new PerlScriptEngineFactory(new PerlProcessBuilderUtilities(),
-                                                                                      new PerlVersionGetter(new PerlProcessBuilderUtilities()));
-
         assertThat(perlScriptEngineFactory.getScriptEngine() instanceof PerlScriptEngine, is(true));
     }
 
     @Test
     public void testThatPerlScriptEngineFactoryReturnsNonNullParameterName() {
-        PerlScriptEngineFactory perlScriptEngineFactory = new PerlScriptEngineFactory(new PerlProcessBuilderUtilities(),
-                                                                                      new PerlVersionGetter(new PerlProcessBuilderUtilities()));
-
         assertThat(perlScriptEngineFactory.getParameter(ScriptEngine.NAME), is(notNullValue()));
     }
 
     @Test
     public void testThatPerlScriptEngineFactoryReturnsNonNullParameterEngineVersion() {
-        PerlScriptEngineFactory perlScriptEngineFactory = new PerlScriptEngineFactory(new PerlProcessBuilderUtilities(),
-                                                                                      new PerlVersionGetter(new PerlProcessBuilderUtilities()));
-
         assertThat(perlScriptEngineFactory.getParameter(ScriptEngine.ENGINE_VERSION), is(notNullValue()));
```

Comment:
If you really want to test getters, you could make the field `parameters` package private in `PerlScriptEngineFactory`. This way, you can compare the return value from `getParameter` with the constants in `parameters`.